### PR TITLE
fix: upgrade swiper package in carousel

### PR DESCRIPTION
**Pull request by gjulivan**
Bug fix on Carousel by upgrading swiper package from 6 to 9.

fix a11y issue.
some DOM changes is occurs.

### CODE REVIEW

**Review by samuelreichert — APPROVED** [review]

File: packages/pluggableWidgets/carousel-web/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased]
 
+### Changed
+
+-   We updated swiper package version to 9.4
```
Comment (r0b1n) [inline packages/pluggableWidgets/carousel-web/CHANGELOG.md:11]: We probably want to mention that there is a noticeable DOM change. Do we want to bump major version, btw, maybe it is worth it.

Comment (gjulivan) [reply packages/pluggableWidgets/carousel-web/CHANGELOG.md:11]: agree

Comment (gjulivan) [reply packages/pluggableWidgets/carousel-web/CHANGELOG.md:11]: as far as i know,
DOM changes without feature improvement can only be considered as minor version. (similar to pop up menu)

File: packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx [hunk 2]
```diff
@@ -20,25 +21,36 @@ export interface CarouselProps {
...
+        }),
+        modules: [A11y],
+        a11y: {
+            enabled: true
+        }
     };
     return (
         <div className={classNames(className, "widget-carousel")} tabIndex={tabIndex}>
-            <SwiperReact {...options} onClick={onClick}>
+            <ReactSwiper tag={"ul"} {...options} onClick={onClick}>
```
Comment (r0b1n) [inline packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx:47]: Looking at the snapshots, this gives the structure that is not quite correct, `ul` should not have `div`s as it's children. I guess it is better to render `.swiper-wrapper` as `ul` and keep `.swiper` as `div`.

```
<ul className="swiper">
    <div className="swiper-wrapper">
        <li className="swiper-slide"><div /></li>
        <li className="swiper-slide"><div /></li>
    </div>
    <div className="swiper-button-prev"/>
    <div className="swiper-button-next"/>
    <div className="swiper-pagination"/>
  </ul>
```

Comment (gjulivan) [reply packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx]: fixed

File: packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx [hunk 3]
```diff
@@ -20,25 +21,36 @@ export interface CarouselProps {
 export function Carousel(props: CarouselProps): ReactElement {
     const { items, pagination, loop, animation, autoplay, delay, navigation, className, tabIndex, onClick } = props;
 
+    const paginationOptions: PaginationOptions | boolean = pagination && {
+        type: "bullets",
+        clickable: true,
+        renderBullet: (index, className) => `<span aria-controls="ctl_${items[index]?.id}" class="${className}"></span>`
```
[the remark this answers is not in the document]
Comment (r0b1n) [reply packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx:27]: Probably a aria-label as well

File: packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx [hunk 4]
```diff
@@ -20,25 +21,36 @@ export interface CarouselProps {
...
+            enabled: true
+        }
     };
     return (
         <div className={classNames(className, "widget-carousel")} tabIndex={tabIndex}>
-            <SwiperReact {...options} onClick={onClick}>
+            <ReactSwiper tag={"ul"} {...options} onClick={onClick}>
                 {items?.map(item => (
-                    <SwiperSlide key={item.id}>{item.content}</SwiperSlide>
+                    <SwiperSlide tag={"li"} key={item.id} id={`ctl_${item.id}`}>
```
[the remark this answers is not in the document]
Comment (gjulivan) [reply packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx]: resolved

File: packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (gjulivan) [reply packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx]: tabIndex is automatically added by a11y module

**Review by samuelreichert — DISMISSED** [review]

**Review by samuelreichert — DISMISSED** [review]

File: packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx [hunk 5]
```diff
@@ -14,31 +21,61 @@ export interface CarouselProps {
...
-    items: Array<{ id: GUID; content?: ReactNode }>;
+    items: CarouselItem[];
 }
 
 export function Carousel(props: CarouselProps): ReactElement {
-    const { items, pagination, loop, animation, autoplay, delay, navigation, className, tabIndex, onClick } = props;
+    const { items, pagination, loop, animation, autoplay, delay, navigation, className, tabIndex, id, onClick } = props;
+    const [swiperRef, setSwiperRef] = useState<SwiperClass>();
+
+    const generateSliderId = useCallback(
```
Comment (r0b1n) [inline packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx:31]: I think something like `getSlideId` is a better name for this callback. I would argue that this function doesn't generate id in a sense of creating something new, also it does it for one slide, not the whole slider.

File: packages/pluggableWidgets/carousel-web/src/Carousel.editorPreview.tsx [hunk 6]
```diff
@@ -10,6 +10,7 @@ export function getPreviewCss(): string {
 export function preview(props: CarouselPreviewProps): ReactNode {
     return (
         <Carousel
+            id={"Carousel"}
```
Comment (r0b1n) [inline packages/pluggableWidgets/carousel-web/src/Carousel.editorPreview.tsx:13]: This possibly might cause id clashing as well in design mode, I would stay safe and use unique id here as well.

File: packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx [hunk 7]
```diff
@@ -14,31 +21,61 @@ export interface CarouselProps {
...
     return (
         <div className={classNames(className, "widget-carousel")} tabIndex={tabIndex}>
-            <SwiperReact {...options} onClick={onClick}>
-                {items?.map(item => (
-                    <SwiperSlide key={item.id}>{item.content}</SwiperSlide>
+            <ReactSwiper wrapperTag={"ul"} {...options} onClick={onClick} onSwiper={setSwiperRef}>
+                {items?.map((item, index) => (
+                    <SwiperSlide
+                        tag={"li"}
+                        aria-hidden={swiperRef?.activeIndex === index}
```
Comment (r0b1n) [inline packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx:71]: So, the slide **is hidden** (`aria-hidden={true}`) when it **is active**. Is it as intended?

Comment (gjulivan) [reply packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx:71]: i was surprise by this condition too, 
but the result was the other way around.
@leonardomendix might be able to confirm this.

File: packages/pluggableWidgets/carousel-web/src/Carousel.editorPreview.tsx (hunk 6 above)
Comment (gjulivan) [reply packages/pluggableWidgets/carousel-web/src/Carousel.editorPreview.tsx]: i think accordion was using the same way.
but nvm, this is fixed using useId

File: packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx (hunk 5 above)
Comment (gjulivan) [reply packages/pluggableWidgets/carousel-web/src/components/Carousel.tsx]: fixed

**Review by r0b1n — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**